adds scope of variable documentation (#1173)
docksal/docksal: docs/content/stack/configuration-variables.md

Configuration variables can be specified in the `docksal.env` file or by running the `fin config set` command (e.g., `fin config set --global PROJECT_INACTIVITY_TIMEOUT="0.5h"`). All variables that are changed from their defaults below will require, at minimum, a project restart `fin restart` so that they can take effect. Additional steps may be required.

> 

Reviewer: ```suggestion

There are 2 scopes of variables: global and project. Global scope variables are supposed to be set in `~/.docksal/docksal.env` while project scope variables should be set in `$PROJECT_ROOT/.docksal/docksal.env`.

```

### DOCKSAL_NFS_PATH

`Default: '/Users'`

**macOS only.** Sets the location of the folder on the host machine to mount to VirtualBox. See [file sharing](/core/file-sharing/) for more information.

### IMAGE_SSH_AGENT (global)

> `Default: docksal/ssh-agent:1.0`

Reviewer: ```suggestion
`Default: docksal/ssh-agent:1.3`
```

The Docker image to use for SSH Agent. This contains the keys that are shared among the projects.

### IMAGE_VHOST_PROXY (global)

> `Default: docksal/vhost-proxy:1.3` 

Reviewer: ```suggestion
`Default: docksal/vhost-proxy:1.5` 
```

Docker image to use as the VHOST Proxy. This forwards all requests to the appropriate container based upon the hostname that is being requested.

### IMAGE_DNS (global)

> `Default: docksal/dns:1.0`

Reviewer: ```suggestion
`Default: docksal/dns:1.1`
```

> 

Reviewer: ```suggestion

`Default: '/Users'`

```

> The location of the folder on the host machine to mount to VirtualBox. See [file sharing](/core/file-sharing/) for more information.

Reviewer: ```suggestion
**macOS only.** Sets the location of the folder on the host machine to mount to VirtualBox. See [file sharing](/core/file-sharing/) for more information.
```

**Warning**: on Mac / Windows, this will open up an SSH proxy over TCP listening on the internal `DOCKSAL_HOST_IP` address.
Other users on your system are able to connect to this address and use your SSH agent.

### DOCKSAL_SSH_PROXY_PORT (global)

Default: `30001`

The TCP port to use to listen on `DOCKSAL_HOST_IP` when proxying SSH authentication to an agent running on the host. Only used
on Mac / Windows if `DOCKSAL_SSH_AGENT_USE_HOST` is set to `1`.

### DOCKSAL_LOCK_UPDATES (global)

`Default: 0`

When set to `1`, this will prevent Docksal from installing and checking for updates. This is usually good in combination with `CI=true`.

### DOCKER_VERSION_ALERT_SUPPRESS (global)

When set, suppresses alert about the outdated Docksal version.

> ### DOCKSAL_ENVIRONMENT (global)

Reviewer: ```suggestion
### DOCKSAL_ENVIRONMENT (project)
```

Note: `DOCKSAL_ENVIRONMENT` should not be set and will not work in the project's `docksal.env` file (this is by design).

### DOCKSAL_VERSION (global)

`Default: master`

Allows for overriding the Docksal version used for checking for updates.

### DOCKSAL_STATS_OPTOUT (global)

`Default: 0` 

Allow for collecting of statistical usage of docksal. When set to `1` this will no longer send statistics.

### DOCKER_NATIVE (global)

> `Default: 0`

Reviewer: ```suggestion
`Default: 0 (on Linux: 1)`
```

For VirtualBox set to `0`.
For Docker Native set to `1`.

### DOCKSAL_DNS_UPSTREAM (global)

Override the default DNS server that Docksal uses. For environments where access to Google DNS server (`8.8.8.8`) is blocked, it should be set to the LAN DNS server. This is often true for VPN users or users behind a corporate firewall.

### DOCKSAL_VHOST_PROXY_IP (global)

Used to set the IP address for the Docksal reverse proxy to listen on. When `CI` variable is set to `true` this will be set to `0.0.0.0`.

### DOCKSAL_DNS_DOMAIN (global)

`Default: docksal`

> This is the domain to use which is tacked on to the end of the projects url.

Reviewer: ```suggestion
This is the domain name used for Docksal projects url, i.e. `http://$PROJECT_NAME.$DOCKSAL_DNS_DOMAIN`. Project named `myproject` will result in `http://myproject.docksal` url by default.
```

**WARNING: do not override this variable unless you know what you are doing.**

### HOST_GID (global)

Group ID for the Container User. On MacOS & Linux defaults to current group account `id -g`.

**WARNING: do not override this variable unless you know what you are doing.**

### XDEBUG_ENABLED (project)

`Default:  0`

Enables PHP XDebug Service for debugging. See [XDebug](/tools/xdebug/).

### SECRET_SSH_PRIVATE_KEY (global)

Use to pass an additional private SSH key. The key is stored in `/home/docker/.ssh/id_rsa` inside `cli` and will be considered by the SSH client in addition to the keys loaded in `docksal-ssh-agent` when establishing a SSH connection from within `cli`.

### SECRET_ACAPI_EMAIL (global or project)

Acquia Cloud API Email Address. See [Acquia Drush Commands](/tools/acquia-drush/).

### SECRET_ACAPI_KEY (global or project)

Acquia Cloud API Key. See [Acquia Drush Commands](/tools/acquia-drush/).

### SECRET_TERMINUS_TOKEN (global or project)

Token used for logging in to Pantheon's CLI Tool [Terminus](/tools/terminus/).

### CLI_USER (global or project)

`Default:  docker`

User to be used for `fin exec`.

> ## Image Variables (project)

Reviewer: ```suggestion
## Image Variables
```

The following variables are used to set images to a specified version. Using them will prevent the images from being udpated
when Docksal is updated.

> ### CLI_IMAGE (project) (project)

Reviewer: ```suggestion
### CLI_IMAGE (project)
```

For VirtualBox set to `0`.
For Docker Native set to `1`.

### DOCKSAL_DNS_UPSTREAM (global)

> 

Reviewer: ```suggestion

`Default: autodetected with fallback to 8.8.8.8`

```

Override the default DNS server that Docksal uses. For environments where access to Google DNS server (`8.8.8.8`) is blocked, it should be set to the LAN DNS server. This is often true for VPN users or users behind a corporate firewall.

### DOCKSAL_VHOST_PROXY_IP (global)

> 

Reviewer: ```suggestion

`Default: 192.168.64.100`

```

Used to set the IP address for the Docksal reverse proxy to listen on. When `CI` variable is set to `true` this will be set to `0.0.0.0`.

### DOCKSAL_DNS_DOMAIN (global)

`Default: docksal`

This is the domain name used for Docksal projects url, i.e. `http://$PROJECT_NAME.$DOCKSAL_DNS_DOMAIN`. Project named `myproject` will result in `http://myproject.docksal` url by default.

### DOCKSAL_NO_DNS_RESOLVER (global)

> 

Reviewer: ```suggestion

`Default: ''`

```
